linkerd/tls/src/imp/boring.rs

```diff
@@ -0,0 +1,243 @@
+use crate::{ClientId, HasNegotiatedProtocol, NegotiatedProtocolRef};
+use linkerd_identity::{ClientConfig, Name, ServerConfig};
+use linkerd_io::{AsyncRead, AsyncWrite, Error, ErrorKind, PeerAddr, ReadBuf, Result};
+use std::net::SocketAddr;
+use std::{
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use std::str::FromStr;
+use tracing::debug;
+use {
+    boring::{
+        ssl,
+        ssl::{
+            SslAcceptor, SslAcceptorBuilder, SslConnector, SslConnectorBuilder, SslMethod,
+            SslVerifyMode,
+        },
+        x509::store::X509StoreBuilder,
+    },
+    tokio_boring::SslStream,
+};
+
+#[derive(Clone)]
+pub struct TlsConnector(ssl::SslConnector);
+
+impl TlsConnector {
+    pub async fn connect<IO>(&self, domain: Name, stream: IO) -> Result<client::TlsStream<IO>>
+    where
+        IO: AsyncRead + AsyncWrite + Unpin,
+    {
+        let conf = self.0.configure().unwrap();
+        match tokio_boring::connect(conf, domain.as_ref(), stream).await {
+            Ok(ss) => Ok(ss.into()),
+            Err(err) => {
+                let _ = err.ssl();
+                Err(Error::new(ErrorKind::Other, "Ble"))
+            }
```

Comment:
This needs some cleaning up

Reply:
Error handling in the boring crate is being revamped here: https://github.com/cloudflare/boring/pull/23

Reply:
I'll revisit that PR at a later time because I have some other stuff on my plate and I misunderstood some critical parts of AsyncRead hah.